### PR TITLE
`yii\db\QueryBuilder::truncateTable` should work consistent over all databases

### PR DESCRIPTION
**Pull request by boboldehampsink**
Cubrid, MSSQL, MySQL, and OCI already restart the identity by default, PGSQL doesn't. This PR makes the use of the `yii\db\QueryBuilder::truncateTable` method more consistent.

_Note: RESTART IDENTITY is available from PGSQL 8.4+_

| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | #13577

### CODE REVIEW

**Comment by samdark** [comment]
The requirement of 8.4+ (2009) is OK. Would be good to mention it in docs here: http://www.yiiframework.com/doc-2.0/guide-db-dao.html

**Review by samdark — APPROVED** [review]

**Comment by boboldehampsink** [comment]
Updated all docs

**Review by SilverFire — APPROVED** [review]

File: framework/db/pgsql/QueryBuilder.php [hunk 1]
```diff
@@ -206,6 +206,16 @@ public function checkIntegrity($check = true, $schema = '', $table = '')
     }
 
     /**
+     * Builds a SQL statement for truncating a DB table.
```
Comment (dynasource) [inline framework/db/pgsql/QueryBuilder.php]: can you add a description for the reason for restarting the Identity?

**Review by dynasource — CHANGES_REQUESTED** [review]

File: framework/db/pgsql/QueryBuilder.php (hunk 1 above)
Comment (boboldehampsink) [inline framework/db/pgsql/QueryBuilder.php]: Sure, what about "Restarts identity to be consistent with other databases."?

Comment (dynasource) [inline framework/db/pgsql/QueryBuilder.php]: could you add an explanation about this 'consistency' and the exception for PGSQL for future reference?

Comment (boboldehampsink) [inline framework/db/pgsql/QueryBuilder.php]: "Explicitly restart identity for PGSQL to be consistent with other databases which all do this by default"

Comment (dynasource) [inline framework/db/pgsql/QueryBuilder.php]: perfect

**Review by dynasource — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**